cohort-engine-api-web/src/test/java/com/ibm/cohort/engine/api/service/vt/DefaultVT.java

```diff
@@ -219,9 +262,70 @@ public void testXMLUnitAssertXPathValues() throws Exception {
+		final String RESOURCE = getUrlBase() + "/{version}/valueset";
+
+		String auth = dataServerConfig.getUser() + ":" + dataServerConfig.getPassword();
+		byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+
+		RequestSpecification request = buildBaseRequest(new Headers(new Header("Authorization", new String(encodedAuth))))
+				.param("version", ServiceBuildConstants.DATE)
+				.queryParam("fhir_server_rest_endpoint", dataServerConfig.getEndpoint())
+				.queryParam("updateIfExists", false)
+				.multiPart(CohortEngineRestHandler.VALUE_SET_PART, new File("src/test/resources/2.16.840.1.113762.1.4.1114.7.xlsx"));
+
+		ValidatableResponse response = request.post(RESOURCE, getServiceVersion()).then();
+		runSuccessValidation(response, ContentType.JSON, HttpStatus.SC_OK);
+	}
+
+	@Test
+	public void testValueSetAlreadyExists(){
+		testValueSetUpload();
+		final String RESOURCE = getUrlBase() + "/{version}/valueset";
+
+		String auth = dataServerConfig.getUser() + ":" + dataServerConfig.getPassword();
+		byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+
+		RequestSpecification request = buildBaseRequest(new Headers(new Header("Authorization", new String(encodedAuth))))
+				.param("version", ServiceBuildConstants.DATE)
+				.queryParam("fhir_server_rest_endpoint", dataServerConfig.getEndpoint())
+				.queryParam("updateIfExists", false)
+				.multiPart(CohortEngineRestHandler.VALUE_SET_PART, new File("src/test/resources/2.16.840.1.113762.1.4.1114.7.xlsx"));
+
+		ValidatableResponse response = request.post(RESOURCE, getServiceVersion()).then();
+		runSuccessValidation(response, ContentType.JSON, 409);
+	}
+
+	@Test
+	public void testValueSetOverride(){
+		testValueSetUpload();
+		final String RESOURCE = getUrlBase() + "/{version}/valueset";
+
+		String auth = dataServerConfig.getUser() + ":" + dataServerConfig.getPassword();
+		byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+
+		RequestSpecification request = buildBaseRequest(new Headers(new Header("Authorization", new String(encodedAuth))))
+				.param("version", ServiceBuildConstants.DATE)
+				.queryParam("fhir_server_rest_endpoint", dataServerConfig.getEndpoint())
+				.queryParam("updateIfExists", true)
+				.multiPart(CohortEngineRestHandler.VALUE_SET_PART, new File("src/test/resources/2.16.840.1.113762.1.4.1114.7.xlsx"));
+
+		ValidatableResponse response = request.post(RESOURCE, getServiceVersion()).then();
+		runSuccessValidation(response, ContentType.JSON, HttpStatus.SC_OK);
+	}
+
 	@Override
 	protected Logger getLogger() {
 		return logger;
 	}
 
+
+	@After
+	public void cleanUp(){
+
+		IGenericClient terminologyClient = FHIRRestUtils.getFHIRClient(dataServerConfig.getEndpoint(), dataServerConfig.getUser(), dataServerConfig.getPassword(), null, null, null, null);
+		ValueSetUtil.deleteValueSet(terminologyClient, "http://cts.nlm.nih.gov/fhir/ValueSet/testValueSet");
```

Comment:
This makes a lot of sense to reset the FHIR server to a known state, but still not sure how those earlier update tests work unless they are relying on data created by the first create test.